python/pyarrow/tests/parquet/test_parquet_encryption.py

```diff
@@ -0,0 +1,388 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pyarrow as pa
+import pytest
+
+try:
+    import pyarrow.parquet as pq
+except ImportError:
+    pq = None
+
+import base64
+from cryptography.fernet import Fernet
+from cryptography.fernet import InvalidToken
+from collections import OrderedDict
+from datetime import timedelta
+
+DATA_TABLE = pa.Table.from_pydict(
+    OrderedDict([
+        ('a', pa.array([1, 2, 3])),
+        ('b', pa.array(['a', 'b', 'c'])),
+        ('c', pa.array(['x', 'y', 'z']))
+    ])
+)
+PARQUET_NAME = 'encrypted_table.in_mem.parquet'
+FOOTER_KEY = Fernet.generate_key()
+FOOTER_KEY_NAME = "footer_key"
+COL_KEY = Fernet.generate_key()
+COL_KEY_NAME = "col_key"
+BASIC_ENCRYPTION_CONFIG = pq.EncryptionConfiguration(
+    footer_key=FOOTER_KEY_NAME,
+    column_keys={
+        COL_KEY_NAME: ["a", "b"],
+    },
+)
+
+
+class InMemoryKmsClient(pq.KmsClient):
+    """This is a mock class implementation of KmsClient, built for testing only.
+    """
+
+    def __init__(self, config):
+        """Create an InMemoryKmsClient instance."""
+        pq.KmsClient.__init__(self)
+        self.master_keys_map = config.custom_kms_conf
+
+    def wrap_key(self, key_bytes, master_key_identifier):
+        """Wrap key key_bytes with key identified by master_key_identifier.
+        The result contains nonce concatenated before the encrypted key."""
+        master_key = self.master_keys_map[master_key_identifier]
+        # Create a cipher object to encrypt data
+        cipher = Fernet(master_key.encode('utf-8'))
```

Comment:
Since this is a basic test class, what does it bring to use `cryptography` here? You can just fake any "cipher".